src/sensors/IMD.py

```diff
@@ -0,0 +1,142 @@
+from pin.pinout import Pinout
+from services.InputCapture import InputCapture
+from services.digital_in import DigitalInService
+from services.digital_out import DigitalOutService
+from shared_memory import SharedMemory
+from src.pin import memory
+from test_lib.input.aggregate import Multiple
+from test_lib.input.input import Input, InputFailedException
+from test_lib.condition.condition import Condition
+from test_lib.input.timed import After
+from enum import Enum
+
+
+class Case(Enum):
+    ShortCircuit = 0
+    Normal = 10
+    UnderVoltage = 20
+    EquipmentFault = 40
+    GroundFault = 50
+
+
+class IMD:
+    def __init__(
+        self, shm: SharedMemory, ok_hs: Pinout, m_hs: Pinout, power_on: Pinout
+    ):
+        self.power_on = DigitalOutService(shm, power_on)
+        self.ok_hs = DigitalInService(shm, ok_hs)
+        self.m_hs = InputCapture(shm, m_hs)
+
+    def _IMDInput(self, frequency, duty_cycle, good_state: bool) -> Input:
+        if self.power_on.get_pin_state() == memory.DigitalOut.State.Low:
+            raise InputFailedException
```

Comment:
Create a more specific exception that inherits from `InputFailedException` and that prints a good error message (implement the `__str__` method to return something like "IMD is not on")